x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/search/aggregations/metrics/GeoShapeCentroidAggregatorTests.java

```diff
@@ -154,10 +153,14 @@ public void testSingleValuedField() throws Exception {
             );
             Geometry geometry = geometryGenerator.apply(false);
             try {
-                geometries.add(GeometryNormalizer.apply(Orientation.CCW, geometry));
-            } catch (InvalidShapeException e) {
-                // do not include geometry
+                geometry = GeometryNormalizer.apply(Orientation.CCW, geometry);
+                // make sure we can index the geometry
+                GeoTestUtils.binaryGeoShapeDocValuesField("field", geometry);
+            } catch (IllegalArgumentException e) {
+                // do not include geometry.
+                continue;
```

Comment:
It doesn't happen too often and it would be nice to know when it happens. So, maybe we should replace it with an assumption instead of silently ignoring it:
```suggestion
                assumeNoException("The geometry[" + geometry.toString() + "] is not supported", e);
```